### PR TITLE
#1170 Extract key extractor & item layout functions

### LINKED ISSUE #1170: Extract keyExtractor and getItemLayout from pages
## Is your feature request related to a problem? Please describe.

Each page uses a `keyExtractor` and `getItemLayout` function. They are exact duplicates everywhere. They should be extracted to their own file.

## Describe the solution you'd like

Extract each to their own file. Not sure exactly which file. `DataTableUtilities/utilities`...?

## Describe alternatives you've considered

Not doing this

## Additional context

@Chris-Petty anything else that should be extracted?

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #1170 

## Change summary

- Adds `getItemLayout` and `keyExtractor` to a utilities file.

Not the greatest factoring, but might prevent a mistake in the future

## Testing

Steps to reproduce or otherwise test the changes of this PR:

*internal change can't test?*

### Related areas to think about

N/A

### CODE REVIEW

File: src/pages/CustomerInvoicePage.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ import { SearchBar } from 'react-native-ui-components';
 
 import { MODAL_KEYS, getModalTitle } from '../utilities';
 import { buttonStrings, modalStrings } from '../localization';
+import { DEFAULT_KEY_EXTRACTOR, DEFAULT_GET_ITEM_LAYOUT } from './dataTableUtilities/utilities';
```
Comment (Chris-Petty) [inline src/pages/CustomerInvoicePage.js]: Looks weird to me to name function variable exports in UPPER_SNAKE_CASE, I'd normally reserve that for sources of string constants. What do you reckon?

Comment (joshxg) [reply src/pages/CustomerInvoicePage.js]: yup I agree - any suggestion though? Obviously landed here cause I was out of ideas. I'll update SI if/when I commit new names 🤣

Comment (Chris-Petty) [reply src/pages/CustomerInvoicePage.js]: Just regular `keyExtractor` or maybe `recordKeyExtractor`/`idExtractor`.

Basically camelCase is fine?

**Comment by joshxg** [comment]
Have updated from #1043, added to SI and renamed

**Pull request merged**